letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -88,20 +82,32 @@ class ApacheConfigurator(augeas_configurator.AugeasConfigurator):
 
     description = "Apache Web Server - Alpha"
 
+    # Kept in same function to avoid multiple compilations of the regex
+
+    private_ips_regex = re.compile(
+        r"(^127\.0\.0\.1)|(^10\.)|(^172\.1[6-9]\.)|"
+        r"(^172\.2[0-9]\.)|(^172\.3[0-1]\.)|(^192\.168\.)")
+    hostname_regex = re.compile(
+        r"^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*[a-z]+$", re.IGNORECASE)
+
+
     @classmethod
     def add_parser_arguments(cls, add):
-        add("server-root", default=constants.CLI_DEFAULTS["server_root"],
-            help="Apache server root directory.")
         add("ctl", default=constants.CLI_DEFAULTS["ctl"],
-            help="Path to the 'apache2ctl' binary, used for 'configtest' and "
-                 "retrieving Apache2 version number.")
+            help="Path to the 'apache2ctl' binary, used for 'configtest', "
+                 "retrieving the Apache2 version number, and initialization "
+                 "parameters.")
         add("enmod", default=constants.CLI_DEFAULTS["enmod"],
             help="Path to the Apache 'a2enmod' binary.")
+        add("dismod", default=constants.CLI_DEFAULTS["dismod"],
```

Comment:
I think you meant a2enmod directly corresponds to a2dismod.

My first implementation of this code just simply checked to see if there was an a2dismod located in the same directory/path and ran that.

I worried about custom setups/deployments and randomly calling out to an unspecified binary that happens to be in the same directory as the "a2enmod" script. This would potentially allow you to run any arbitrary program as root.

I realize that this is making fairly lax assumptions on the naming conventions/directory layout. I also would worry about the names changing slightly in the future.